crypto/evp/p_lib.c

```diff
@@ -2084,7 +2087,7 @@ void *evp_pkey_get_legacy(EVP_PKEY *pk)
         return ret;
 
     if (!evp_pkey_copy_downgraded(&tmp_copy, pk))
-        return NULL;
+        goto err;
```

Comment:
This doesn't look right.
There is something wrong with evp_pkey_copy_downgraded() if it returns tmp_copy on error..

Reply:
It can allocate a new BN even in case of failure. And all the other callers of evp_pkey_copy_downgraded() handle that. So rather than changing the function it seems to me much more logical to change the last caller that does not handle that case.

Reply:
ok.. but it is not a good pattern

Reply:
I agree, but fixing it would imo be refactoring rather than a bug fix.

Reply:
Isn't this #18470?

Reply:
Heh, it is. I did not realize that :grin:

Reply:
Anyway it does not conflict with #18470 so perhaps we can keep this as well?